tests/flytekit/unit/core/test_type_hints.py

```diff
@@ -1338,3 +1340,16 @@ def t2() -> dict:
     expected_struct = Struct()
     expected_struct.update({"k1": "v1", "k2": 3, "4": {"one": [1, "two", [3]]}})
     assert output_lm.literals["o0"].scalar.generic == expected_struct
+
+
+def test_error_messages():
```

Comment:
can we have a test where the mismatch happens in a proto? We had an issue in the [OSS slack ](https://flyte-org.slack.com/archives/CREL4QVAQ/p1631818531170100?thread_ts=1630642686.070000&cid=CREL4QVAQ) where a workflow input of type Dictionary but it was being passed a list of dictionaries (and as you can see from the slack message, the error is not intuitive).